hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1285,7 +1293,7 @@ private void sendMembershipEventNotifications(MemberImpl member, Set<Member> mem
             final MembershipServiceEvent event = new MembershipServiceEvent(membershipEvent);
             for (final MembershipAwareService service : membershipAwareServices) {
                 // service events should not block each other
-                nodeEngine.getExecutionService().execute(ExecutionService.SYSTEM_EXECUTOR, new Runnable() {
+                nodeEngine.getExecutionService().execute(MEMBERSHIP_EVENT_EXECUTOR_NAME, new Runnable() {
```

Comment:
correct, we shouldn't let events drop. by advising event-executor I was aiming to use `StripedRunnable` but in this case it is not safe to use.